### PR TITLE
HTTPs in ProxyServerAssetReader

### PR DESCRIPTION
**Pull request by grouma**
Closes https://github.com/dart-lang/webdev/issues/1112

### CODE REVIEW

File: dwds/lib/src/readers/proxy_server_asset_reader.dart [hunk 1]
```diff
@@ -23,13 +23,15 @@ class ProxyServerAssetReader implements AssetReader {
...
+    isHttps ??= false;
+    var scheme = isHttps ? 'https://' : 'http://';
     var client = isHttps
         ? IOClient(
             HttpClient()..badCertificateCallback = (cert, host, port) => true)
         : null;
-    _handler =
-        proxyHandler('$scheme$host:$assetServerPort/$root', client: client);
+    var url = '$scheme$host:$assetServerPort/';
+    if (root != null) url += '$root/';
```
Comment (jacob314) [inline dwds/lib/src/readers/proxy_server_asset_reader.dart:33]: nit: add curly braces? I thought the one dart exception was `return`

**Review by jacob314 — APPROVED** [review]
lgtm

File: dwds/lib/src/readers/proxy_server_asset_reader.dart (hunk 1 above)
Comment (grouma) [reply dwds/lib/src/readers/proxy_server_asset_reader.dart:33]: Effective Dart says you can ommit braces if everything fits on a single line:
https://dart.dev/guides/language/effective-dart/style#do-use-curly-braces-for-all-flow-control-structures

**Comment by grouma** [comment]
Merging on red as head is broken.

**Pull request merged**